src/main/java/de/muenchen/isi/domain/service/AbfrageStatusService.java

```diff
@@ -302,4 +299,116 @@ public UUID getAbfrageId(final MessageHeaders messageHeaders) throws EntityNotFo
         }
         return abfrageId;
     }
+
+    /**
+     * Ermittelt mögliche Statusübergänge basierend auf Benutzer Authorities und aktuellen Status der Abfrage.
+     *
+     * @param id vom Typ {@link UUID} um die Abfrage zu finden
+     * @return Liste von {@link TransitionModel} welche möglich sind
+     * @throws EntityNotFoundException falls die Abfrage nicht gefunden wird
+     */
+    public List<TransitionModel> getStatusAbfrageEventsBasedOnStateAndAuthorities(final UUID id)
+        throws EntityNotFoundException {
+        List<AuthoritiesEnum> authorities = AuthenticationUtils.getUserAuthorities();
+        List<StatusAbfrageEvents> possibleAbfrageEventsBasedOnAuthorities;
+        Map<AuthoritiesEnum, StatusAbfrageEvents> authoritiesAndMatchingEvent = getAuthoritiesAndEventsMap();
+        possibleAbfrageEventsBasedOnAuthorities =
+            getStatusAbfrageEventsForAuthorities(authorities, authoritiesAndMatchingEvent);
+        List<StatusAbfrageEvents> matchingAbfrageEvents = new ArrayList<>(getStatusAbfrageEventsBasedOnState(id));
+        matchingAbfrageEvents.retainAll(possibleAbfrageEventsBasedOnAuthorities);
+        return matchingAbfrageEvents
+            .stream()
+            .map(event -> {
+                TransitionModel transitionModel = new TransitionModel();
+                transitionModel.setUrl(event.getUrl());
+                transitionModel.setButtonName(event.getButtonName());
+                transitionModel.setIndex(event.getIndex());
+                transitionModel.setDialogText(event.getDialogText());
+                return transitionModel;
+            })
+            .sorted(Comparator.comparingInt(TransitionModel::getIndex))
+            .collect(Collectors.toList());
+    }
+
+    /**
+     * Ermittelt anhand der vom Benutzer besessenen Authorities, welche {@link StatusAbfrageEvents} für den Benutzer erlaubt sind.
+     *
+     * @param authorities                 Die Liste der Authorities des Benutzers.
+     * @param authoritiesAndMatchingEvent Die {@link Map}, die die Zuordnung von Authorities zu den dazugehörigen {@link StatusAbfrageEvents} enthält.
+     * @return Eine Liste der erlaubten {@link StatusAbfrageEvents} basierend auf den Authorities des Benutzers.
+     */
+    private List<StatusAbfrageEvents> getStatusAbfrageEventsForAuthorities(
+        final List<AuthoritiesEnum> authorities,
+        Map<AuthoritiesEnum, StatusAbfrageEvents> authoritiesAndMatchingEvent
+    ) {
+        List<StatusAbfrageEvents> statusAbfrageEventsList = new ArrayList<>();
+        for (AuthoritiesEnum authority : authorities) {
```

Comment:
Evtl in einem Stream abbilden. Streams lesen sich meistens leichter als for oder while-Loops.
Evtl.:
```java
        return authorities.stream()
                .map(authoritiesAndMatchingEvent::get)
                .filter(Objects::nonNull)
                .collect(Collectors.toList());

```